test/test_resize.cpp

```diff
@@ -3326,4 +3326,81 @@ TEST_F(ResizeTest, NoncancellableSplitAndCat_CatOnlySubsetOfSplitOutputs) {
   EXPECT_FALSE(out_tensors[0].is_alias_of(in_tensor));
 }
 
+TEST_F(ResizeTest, CancellableSplitAndCat_PermuteInBetween) {
```

Comment:
Can we add a test where we have:

```
TensorView* in = makeContigConcreteTensor({2, 2, 4});
TensorView* s0 = broadcast(slice(in, {0, 0, 0}, {2, 2, 2}), {false, true, false, false});
TensorView* s1 = broadcast(slice(in, {0, 0, 2}, {2, 2, 4}), {false, true, false, false});
TensorView* out = cat({s0, s1, s2}, /*dim=*/2);
```

Reply:
Done. I assume this is to verify the optimization correctly bails out when seeing an unsupported op between the split and the cat?

Reply:
Thanks for adding the test. I was wrong about the horizontal merge logic, but I'm happy with adding that test.